CE#1790

sparksam opened the pull request:
CE#1790: Added the possibility to create a repo automatically in Github and push the Github repo migrated book to Github.

sparksam committed:
CE#1790: Added the possibility to create a repo automatically in Github and push the Github repo migrated book to Github.

sparksam committed:
CE#1790: Updated the Commit message and local user name

sparksam committed:
CE#1790: added a possibility to clone a parent repository, checkout a branch for the derived book. Generate a github template for the book and push to new repository.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Added Python scripts to query CNX archive DB, generate JSON files and update HTML table. Added the generated JSON files and index.html file.
Updated the Readme.md
Removed unused code.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Added Python scripts to query CNX archive DB, generate JSON files and update HTML table. Added the generated JSON files and index.html file.
Updated the Readme.md
Removed unused code.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Possibilty to create user and organizations repositories.
Added the fetch all option to dowload all parent remote repos locally before pushing.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Possibilty to create user and organizations repositories.
Added the fetch all option to dowload all parent remote repos locally before pushing.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Possibilty to create user and organizations repositories.
Added the fetch all option to dowload all parent remote repos locally before pushing.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Added Slack notification to ce-stream

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Built and pushed a docker image that contains the dependencies and updated the pipelines.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Built and pushed a docker image that contains the dependencies and updated the pipelines.

sparksam committed:
CE#1790, CE#1791 , CE#1792:
Renamed  vendors occurrences to non openstax users.

m1yag1 commented on repository-creator/Dockerfile:
```diff
@@ -0,0 +1,51 @@
+FROM python:3.7-slim-bullseye as base
```
You could also use [`openstax/python3-base:latest`](https://hub.docker.com/layers/158312527/openstax/python3-base/latest/images/sha256-42d4bb38c2ae6ab345c1fa3a692e50f19f4881779893b635cb60267491546bd9?context=repo) which is python 3.8 bullseye unless you need python 3.7 specifically.

sparksam replied:
When I try `python 3.8`, I get some errors installing lxml and neb dependencies.

m1yag1 replied:
ah ok. i think i vaguely remember something about that.. no worries then.

sparksam committed:
Debug book migration failure in concourse prod.

sparksam committed:
Debug book migration failure in concourse prod. (1)

sparksam committed:
Add conditions to check Github rate limits.

sparksam committed:
Removed the certificates creation from the sync script.

sparksam committed:
Move Github repository check before neb tasks.

sparksam committed:
Updated the pipeline to remove unnecessary debug outputs.

sparksam committed:
Fixed a bug related to empty repos being created.

sparksam committed:
Fixed a bug for existing repositories.

sparksam committed:
Fixed a bug for existing repositories.

sparksam committed:
Fixed a bug for existing repositories. (2)

sparksam committed:
Fixed a bug for existing repositories. (3)

sparksam committed:
Add a check for repository sizes. Updated git to use http instead of ssh urls.

sparksam committed:
Changed the return for repository already exists to successful.

sparksam committed:
Avoid creating repositories when they exist.

sparksam committed:
Fixed an issue related to overridding titles.

 File "/tmp/build/0eed810b/content-synchronizer/resource-synchronizer/override_module_titles.py", line 42, in override_title
    )[0].text
IndexError: list index out of range

sparksam committed:
Added functionality to create PRs for duplicates.

sparksam committed:
Reduce the repository name to only the first 100 characters.

sparksam committed:
Fixed collection.xml file download.

sparksam committed:
Fixed collection.xml file download. (2)

sparksam committed:
Fixed xpath xmlstarlet search

sparksam committed:
Updated .git directory exemption from delete

sparksam committed:
Updated .git directory exemption from delete (1)

sparksam committed:
Updated .git directory exemption from delete (2)

sparksam committed:
Added feature to fork parent repositories.

sparksam committed:
Removed collection_xml from prs and fixed .vscode deletion

sparksam committed:
Removed collection_xml from prs and fixed .vscode deletion (2)

sparksam committed:
Set origin with credentials for derived books migration.

sparksam committed:
Set origin with credentials for derived books migration (1)

sparksam committed:
Set origin with credentials for derived books migration (2)

sparksam committed:
Updated the sync script for openstax forked repositories.

sparksam committed:
Updated the sync script for openstax forked repositories. (2)

sparksam committed:
removed osbdev prefix from repository names.

sparksam committed:
Updated script for derived users books
